### PR TITLE
WIP: WHATWG spec compliance

### PR DESCRIPTION
**Pull request by stevenvachon**
This is not quite ready for merging, but it needs a review. Here is the current progress:

* relative URLs can no longer be supported
* decoding query strings is not a good idea, example: `http://host/?var1=va%26lue&var2=value` becomes `http://host/?var1=va&lue&var2=value`
  * we could try decoding per parameter via `searchParams` and re-encode `=` and `&` characters, but there may be edge cases
* decoding IDNAs is possible but might be a bit much to depend on [tr46](https://npmjs.com/tr46)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 sudo: false
 language: node_js
 node_js:
-  - '6'
-  - '4'
+  - 8
+  - 6
```
Comment (sindresorhus) [inline .travis.yml]: Should be quoted. Versions are not numbers.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "url": "sindresorhus.com"
   },
   "engines": {
-    "node": ">=4"
+    "node": ">=6"
```
Comment (sindresorhus) [inline package.json]: I'm supporting Node.js 4 out its LTS period.

File: package.json [hunk 3]
```diff
@@ -37,9 +37,9 @@
     "canonical"
   ],
   "dependencies": {
+    "minurl": "stevenvachon/minurl",
```
Comment (sindresorhus) [inline package.json]: Use the npm dependency.

**Comment by sindresorhus** [comment]
> decoding query strings is not a good idea, example: `http://host/?var1=va%26lue&var2=value` becomes `http://host/?var1=va&lue&var2=value`

We need a way to get to a canonical version though. What if we have two identical URLs, one with encoded params and one without. They would not normalize to the same URL.

> decoding IDNAs is possible but might be a bit much to depend on [tr46](https://npmjs.com/tr46)

This is also important for the above reason.

File: package.json (hunk 3 above)
Comment (stevenvachon) [inline package.json]: When it's published, it will.

File: package.json (hunk 2 above)
Comment (stevenvachon) [inline package.json]: The shim ([whatwg-url](https://npmjs.com/whatwg-url)) is 6+

**Comment by sindresorhus** [comment]
Closing in favor of https://github.com/sindresorhus/normalize-url/pull/59

**Pull request closed without merging**